fluent-plugin-events/test/plugin/test_in_events.rb

```diff
@@ -1,17 +1,99 @@
 require "helper"
 require "fluent/plugin/in_events.rb"
+require 'fluent/test/driver/input'
+require 'fluent/test/log'
 
 class EventsInputTest < Test::Unit::TestCase
-  setup do
-    Fluent::Test.setup
-  end
+  include SumoLogic::Kubernetes::Connector
 
-  test "failure" do
+  def setup
+    # runs before each test
+    @api_version = 'v1'
+    init_globals
+    connect_kubernetes
   end
 
-  private
+  def teardown
+    # runs after each test
+  end
 
+  def log
+    Fluent::Test::TestLogger.new
+  end
+  
   def create_driver(conf)
-    Fluent::Test::Driver::Input.new(Fluent::Plugin::EventsInput).configure(conf)
+    driver = Fluent::Test::Driver::Input.new(Fluent::Plugin::EventsInput).configure(conf)
+  end
+
+  test 'pull_resource_version correctly from eventlist' do
+    config = %([])
+    driver = create_driver(config).instance
+    driver.instance_variable_set(:@client, @client)
+
+    mock_get_events
+    resource_version = driver.pull_resource_version
+    assert_equal resource_version, '2346293'
+  end
+
+  test 'initialize_resource_version correctly for different resources' do
+    config = %([])
+    driver = create_driver(config).instance
+    driver.instance_variable_set(:@client, @client)
+
+    mock_get_config_map
+    resource = driver.initialize_resource_version
+    assert_equal resource.data['resource-version-events'], 'dummy-events-rv'
+    assert_equal resource.data['resource-version-pods'], 'dummy-pods-rv'
+    assert_equal resource.data['resource-version-services'], 'dummy-services-rv'
+  end
+
+  test 'watch_events with default type_selector' do
+    config = %([])
+    driver = create_driver(config).instance
+    driver.instance_variable_set(:@client, @client)
+    mock_watch_events
+
+    selected_events_count = get_watch_resources_count_by_type_selector(["ADDED", "MODIFIED"],
+      'api_watch_events_v1.txt')
+    driver.router.expects(:emit).times(selected_events_count).with(anything, anything, anything)
+    events = driver.start_watcher_thread
+    sleep 5
```

Comment:
Is there some way to do `eventually` in these tests rather than `sleep 5`? I assume it's probably fine to leave it as is, but unless it takes a long time to change I think `eventually` might be better

Reply:
`eventually` is really a good idea, however I didn't find something useful that can work in that way. Or maybe I missed something.
Typically, `expects(:method).times(...)` should work. However, it is way too flaky for this test case. And I chose to `sleep 5` to make sure it does verify `emit` is called. Not sure if things can get better/less flaky if I reduced the volume of events.